### DISCUSSION TITLE
remove old apache-httpclient

### DISCUSSION THREAD

**Discussion by Hugolarson** [vaadin/flow#17564] [General]
Hello,

Please consider move away from the old apache-httpclient 4.x. 
It can be replaced by modern java 11 java.net.http.HttpClient and with it gone you can also get rid of the ancient commons libraries.

All these dependencies are obsolete with modern Java version (11++).

**Comment by mshabarov** [comment]
Thanks @Hugolarson for pointing this out.
I've opened this ticket for Flow https://github.com/vaadin/flow/issues/17734.

**Comment by mshabarov** [comment]
Already implemented in Vaadin 24.2 with https://github.com/vaadin/flow/issues/14538.
I think we can close this discussion then.
